File: Proiect_Adapost/Models/Animal/Animal.cs
using Proiect_Adapost.Models.Base;

namespace Proiect_Adapost.Models.Animal
{
    public class Animal: BaseEntity
    {
        
        public string Nume { get; set; }
        public string Tip { get; set; }
        public string Rasa { get; set; }
        public string Culoare { get; set; }

        public Guid? AdapostId { get; set; }

   
        public Carnet_sanatate.Carnet_sanatate? Carnet_sanatate { get; set; }

        public ICollection<AnimalClient.AnimalClient>? AnimaleClienti { get; set; }

        public Adapost.Adapost? Adapost { get; set; }

    }
}

File: Proiect_Adapost/adapost/src/Animal.ts
import { Adapost } from "./Adapost";
import { Carnet_Sanatate } from "./Carnet_Sanatate";
import { AnimalClient } from "./AnimalClient";

export interface Animal {
    id: string;
    nume: string;
    tip: string;
    rasa: string;
    culoare: string;
    Adapost: Adapost;
    Carnet_Sanatate: Carnet_Sanatate;
    AnimalClients: AnimalClient[];
    }